exporting/send_internal_metrics.c

```diff
@@ -3,16 +3,170 @@
+
+        // ------------------------------------------------------------------------
+
+        snprintf(id, RRD_ID_LENGTH_MAX, "exporting_%s_thread_cpu", instance->config.name);
+        netdata_fix_chart_id(id);
+
+        stats->st_rusage = rrdset_create_localhost(
+            "netdata", id, NULL, buffer_tostring(family), NULL, "Netdata Exporting Instance Thread CPU Usage",
+            "milliseconds/s", "exporting", NULL, 130640, instance->config.update_every, RRDSET_TYPE_STACKED);
+
+        stats->rd_user   = rrddim_add(stats->st_rusage, "user", NULL, 1, 1000, RRD_ALGORITHM_INCREMENTAL);
+        stats->rd_system = rrddim_add(stats->st_rusage, "system", NULL, 1, 1000, RRD_ALGORITHM_INCREMENTAL);
+
+        buffer_free(family);
+
+        stats->initialized = 1;
+    }
+
+    // ------------------------------------------------------------------------
+    // update the monitoring charts
+
+    if (likely(stats->st_metrics->counter_done))
+        rrdset_next(stats->st_metrics);
+
+    rrddim_set_by_pointer(stats->st_metrics, stats->rd_buffered_metrics, stats->buffered_metrics);
+    rrddim_set_by_pointer(stats->st_metrics, stats->rd_lost_metrics,     stats->lost_metrics);
+    rrddim_set_by_pointer(stats->st_metrics, stats->rd_sent_metrics,     stats->sent_metrics);
+
+    rrdset_done(stats->st_metrics);
+
+    // ------------------------------------------------------------------------
+
+    if (likely(stats->st_bytes->counter_done))
+        rrdset_next(stats->st_bytes);
+
+    rrddim_set_by_pointer(stats->st_bytes, stats->rd_buffered_bytes, stats->buffered_bytes);
+    rrddim_set_by_pointer(stats->st_bytes, stats->rd_lost_bytes,     stats->lost_bytes);
+    rrddim_set_by_pointer(stats->st_bytes, stats->rd_sent_bytes,     stats->sent_bytes);
+    rrddim_set_by_pointer(stats->st_bytes, stats->rd_received_bytes, stats->received_bytes);
+
+    rrdset_done(stats->st_bytes);
+
+    // ------------------------------------------------------------------------
+
+    if (likely(stats->st_ops->counter_done))
+        rrdset_next(stats->st_ops);
+
+    rrddim_set_by_pointer(stats->st_ops, stats->rd_transmission_successes, stats->transmission_successes);
+    rrddim_set_by_pointer(stats->st_ops, stats->rd_data_lost_events,       stats->data_lost_events);
+    rrddim_set_by_pointer(stats->st_ops, stats->rd_reconnects,             stats->reconnects);
+    rrddim_set_by_pointer(stats->st_ops, stats->rd_transmission_failures,  stats->transmission_failures);
+    rrddim_set_by_pointer(stats->st_ops, stats->rd_receptions,             stats->receptions);
+
+    rrdset_done(stats->st_ops);
+
+    // ------------------------------------------------------------------------
+
+    struct rusage thread;
+    getrusage(RUSAGE_THREAD, &thread);
+
+    if (likely(stats->st_rusage->counter_done))
+        rrdset_next(stats->st_rusage);
+
+    rrddim_set_by_pointer(stats->st_rusage, stats->rd_user,   thread.ru_utime.tv_sec * 1000000ULL + thread.ru_utime.tv_usec);
```

Comment:
Here you are multiplying the seconds by `1000000ULL` and you sum the result to microseconds, please, can you confirm that you did not want multiply the microseconds by 10 seconds?

Reply:
No. The result in microseconds and is divided by 1000 to get microseconds in the chart.